lib/block_view_message_test.go

```diff
@@ -2392,13 +2558,410 @@ func TestGroupMessages(t *testing.T) {
+		require.NoError(err)
+		messages, _, err = utxoView.GetMessagesForUser(senderPkBytes, chain.blockTip().Height+1)
+		require.NoError(err)
+		require.Equal(7, len(messages))
+		messages, _, err = utxoView.GetMessagesForUser(recipientPkBytes, chain.blockTip().Height+1)
+		require.NoError(err)
+		require.Equal(8, len(messages))
+		messages, _, err = utxoView.GetMessagesForUser(m1PubKey, chain.blockTip().Height+1)
+		require.NoError(err)
+		require.Equal(4, len(messages))
+		messages, _, err = utxoView.GetMessagesForUser(m0PubKey, chain.blockTip().Height+1)
+		require.NoError(err)
+		require.Equal(5, len(messages))
+		messages, _, err = utxoView.GetMessagesForUser(m2PubKey, chain.blockTip().Height+1)
+		require.NoError(err)
+		require.Equal(2, len(messages))
+
+		{
+			// Let us now try to unmute m0 AGAIN
+			// This should produce an error since m0 is already unmuted
+			var unmuteList []*MessagingGroupMember
+			unmuteList = append(unmuteList, &MessagingGroupMember{
+				m0PublicKey,
+				BaseGroupKeyName(),
+				encrypt(privBytes, m0PubKey),
+			})
+			//require.Equal(false, _verifyMessagingKey(testMeta, senderPublicKey, entry))
+			extraData = make(map[string][]byte)
+			extraData[MessagingGroupOperationType] = []byte{byte(MessagingGroupOperationUnmuteMembers)}
+			_messagingKeyWithExtraDataWithTestMeta(
+				testMeta,
+				senderPkBytes,
+				senderPrivString,
+				entry.MessagingPublicKey[:],
+				gangKey,
+				[]byte{},
+				unmuteList,
+				extraData,
+				RuleErrorMessagingMemberAlreadyUnmuted)
+		}
+
+		{
+			// Let us now try to unmute m1 who is not part of the group
+			// This should produce an error since m1 does not exist in the group
+			var unmuteList []*MessagingGroupMember
+			unmuteList = append(unmuteList, &MessagingGroupMember{
+				m1PublicKey,
+				BaseGroupKeyName(),
+				encrypt(privBytes, m1PubKey),
+			})
+			//require.Equal(false, _verifyMessagingKey(testMeta, senderPublicKey, entry))
+			extraData = make(map[string][]byte)
+			extraData[MessagingGroupOperationType] = []byte{byte(MessagingGroupOperationUnmuteMembers)}
+			_messagingKeyWithExtraDataWithTestMeta(
+				testMeta,
+				senderPkBytes,
+				senderPrivString,
+				entry.MessagingPublicKey[:],
+				gangKey,
+				[]byte{},
+				unmuteList,
+				extraData,
+				RuleErrorMessagingMemberNotInGroup)
+		}
+
+		{
+			// MORE MUTING TESTS
+			// Let us now try to mute group owner "sender" as a sanity check
+			// This should fail because GroupOwner cannot mute/unmute herself
+			var muteList []*MessagingGroupMember
+			muteList = append(muteList, &MessagingGroupMember{
+				senderPublicKey,
+				BaseGroupKeyName(),
+				encrypt(privBytes, senderPkBytes),
+			})
+			//require.Equal(false, _verifyMessagingKey(testMeta, senderPublicKey, entry))
+			extraData := make(map[string][]byte)
+			extraData[MessagingGroupOperationType] = []byte{byte(MessagingGroupOperationMuteMembers)}
+			_messagingKeyWithExtraDataWithTestMeta(
```

Comment:
Can we add tests for muting/unmuting oneself? I think that's a different rule error than the group owner one

Reply:
Only owner can mute/unmute people. There was an edge-case where the owner tried muting someone who wasn't in the group. Keshav found this ln and I just fixed it too rn.